tests/test_schema.py

```diff
@@ -581,6 +582,11 @@ def test_invalid_url():
     assert 'homepage' in s.errors
     assert 'Not a valid URL.' in s.errors['homepage'][0]
 
+def test_invalid_dict_but_okay():
+    u = User('Joe', various_data='baddict')
```

Comment:
Yes, this is the expected behavior. In practice, this will not happen, because only data are expected to be validated *before* serialization.